File: app/src/main/java/com/sankar/kotlinmvvm/retrofit/ApiRequest.java
package com.sankar.kotlinmvvm.retrofit;

import com.sankar.kotlinmvvm.response.ListViewResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiRequest {

    @GET("v2/everything/")
    Call<ListViewResponse> getMovieArticles(
            @Query("q") String query,
            @Query("apikey") String apiKey
    );
}

File: app/src/main/java/com/sankar/kotlinmvvm/retrofit/ApiRequest.kt
package com.sankar.kotlinmvvm.retrofit

import com.sankar.kotlinmvvm.model.Model
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiRequest {
    @GET("v2/everything/")
    fun getMovieArticles(
        @Query("q") query: String?,
        @Query("apikey") apiKey: String?
    ): Call<Model?>?
}